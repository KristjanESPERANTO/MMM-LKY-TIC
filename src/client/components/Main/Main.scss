/** @define main */

/**
 * MagicMirror
 * Style rules for Main component
 */

.MMMLKYTICWrapper {
  .main {}
}
