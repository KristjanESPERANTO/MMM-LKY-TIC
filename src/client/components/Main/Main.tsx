import { FunctionComponent } from 'react';
import classnames from 'classnames';
import Teleinfo from '../Teleinfo/Teleinfo';

import '../../styles/reset.scss';
import '../../styles/module.scss';
import './Main.scss';

/**
 * MagicMirror
 * Main REACT component
 */
const Main: FunctionComponent = () => {
  return (
    <div className={classnames('main', 'dimmed', 'light', 'small')}>
      <Teleinfo />
    </div>
  );
}

export default Main;
