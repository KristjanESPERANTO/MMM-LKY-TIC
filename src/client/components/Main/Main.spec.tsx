import renderer from 'react-test-renderer';
import ConfigurationContext from '../../contexts/ConfigurationContext';
import Main from './Main';

jest.mock('../Teleinfo/Teleinfo', () => (props: object) => (
  <div {...props}>Teleinfo component</div>
));

describe('Main component', () => {
  it('should render correctly with default props', () => {
    const tree = renderer.create(<Main />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with custom configuration', () => {
    const tree = renderer
      .create(
        <ConfigurationContext.Provider value={{ currencySymbol: 'S' }}>
          <Main />
        </ConfigurationContext.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
