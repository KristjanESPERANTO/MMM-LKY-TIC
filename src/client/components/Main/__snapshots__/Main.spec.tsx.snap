// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render correctly with default props 1`] = `
<div
  className="main dimmed light small"
>
  <div>
    Teleinfo component
  </div>
</div>
`;
